### PR TITLE
Fixed bug when notice happened when using error_reporting -1

### PR DESCRIPTION
**Pull request by resurtm**
How to reproduce bug and test this fix:
1. Set `error_reporting` to the 0 value and run unit tests. Everything will work fine.
2. Set `error_reporting` to the -1 (same as ~1) value and run unit tests. Notice will happen.

Call stack of the notice when `error_reporting` is -1 and without this fix:

```
Array to string conversion

C:\_work\jetbrains\with-related-behavior\WithRelatedBehavior.php:38
C:\_work\jetbrains\with-related-behavior\WithRelatedBehavior.php:91
C:\_work\jetbrains\with-related-behavior\tests\unit\WithRelatedBehaviorTest.php:40
```

Passes unit tests.

### CODE REVIEW

**Comment by creocoder** [comment]
Thanks for working on this.

File: WithRelatedBehavior.php [hunk 1]
```diff
@@ -35,12 +35,18 @@ public function validate($data=null,$clearErrors=true,$owner=null)
 		else
 		{
 			$attributeNames=$owner->attributeNames();
-			$attributes=array_intersect($data,$attributeNames);
+			// array_intersect must not be used here because when error_reporting is -1 notice will happen
+			// since $data array contains not just scalar string values
+			$attributes=array_uintersect($data,$attributeNames,
+				create_function('$x,$y','return !is_string($x) || !is_string($y) ? -1 : strcmp($x,$y);'));
```
Comment (samdark) [inline WithRelatedBehavior.php:41]: 5.3+?

Comment (resurtm) [inline WithRelatedBehavior.php:41]: Works fine on 5.3+. I've tested it on 5.4.3.

Or you mean we should change it to the real lambda? :-)

**Comment by samdark** [comment]
Nope, it's fine as is.

**Pull request merged**